### PR TITLE
Feature: support mysql geometry function centroid/st_centroid

### PR DESCRIPTION
**Pull request by Young-Cody**
### Task Description
#1399: Support MySQL Spatial Analysis Function [Centroid/ST_Centroid](https://dev.mysql.com/doc/refman/8.0/en/gis-polygon-property-functions.html#function_st-centroid).

### Passed Regressions

It has passed mysql test [spatial_analysis_functions_centroid](https://github.com/mysql/mysql-server/blob/trunk/mysql-test/suite/gis/t/spatial_analysis_functions_centroid.test).
Screenshot from 2023-10-10 09-53-41

### Other Information

- I modified the file [ob_item_type.h](https://github.com/oceanbase/oceanbase/blob/283f3e69281c70cd2174ef344c22a1ccc2f43d15/src/objit/include/objit/common/ob_item_type.h) to reserve more space for gis functions. I don't know if it's ok.

### CODE REVIEW

File: deps/oblib/src/lib/geo/ob_geo_func_utils.h [hunk 1]
```diff
@@ -49,7 +49,8 @@ class ObGeoFuncUtils
                            GcType *gc,
                            typename GcTreeType::sub_mpt_type *&multi_point,
                            typename GcTreeType::sub_ml_type *&multi_line,
-                           typename GcTreeType::sub_mp_type *&multi_poly);
+                           typename GcTreeType::sub_mp_type *&multi_poly,
```
Comment (nautaa) [inline deps/oblib/src/lib/geo/ob_geo_func_utils.h:52]: remove trailing whitespace

**Comment by nautaa** [comment]
Thanks for your contribution. This feature has been developed internally and will be released in the future, so this PR will not continue to merge. If you wish to continue other contributions, please contact us. Thanks again for your contribution!

**Comment by hnwyllmm** [comment]
Sorry for the big mistake.

**Comment by Young-Cody** [comment]
That's all right. I've learned a lot.

**Pull request closed without merging**